components/merkledb/src/views/metadata.rs

```diff
@@ -172,21 +172,30 @@ where
     T: IndexAccess,
     V: BinaryAttribute + Copy + Default,
 {
-    let index_name = index_address.fully_qualified_name();
+    // Actual name.
+    let index_name = index_address.name.clone();
+    // Full name for internal usage.
+    let index_full_name = index_address.fully_qualified_name();
 
     let mut pool = IndexesPool::new(index_access.clone());
-    let (metadata, is_new) = if let Some(metadata) = pool.index_metadata(&index_name) {
+    let (metadata, is_new) = if let Some(metadata) = pool.index_metadata(&index_full_name) {
         assert_eq!(
             metadata.index_type, index_type,
             "Index type does not match specified one"
         );
         (metadata, false)
     } else {
-        (pool.create_index_metadata(&index_name, index_type), true)
+        (
+            pool.create_index_metadata(&index_full_name, index_type),
```

Comment:
What if change signature of `create_index_metadata` to:
```rust
pool.create_index_metadata(&index_name, &index_full_name, index_type) 
```
for doing this assignment `index_address.name = index_name;` there 
???

Reply:
I don't think that it makes sense: `index_name` isn't actually stored in the metadata (only as a part of `full_name`), so it will be weird to store something in structure but not use it.

Reply:
Then it's ok if so.